### DISCUSSION TITLE
Peek last global mark

### DISCUSSION THREAD

**Discussion by luiz00martins** [LeonHeidelbach/trailblazer.nvim#14] [Q&A] [answered]
I want to do the exact same thing as `:TrailBlazerTrackBack`, but without removing the mark. That is:

> Move to the last global trail mark or the last one within the specified buffer, **without removing** it from the trail mark stack.

It would also be nice to know how to move back forward. My objective is to move back and forward freely through the stack.

**Comment by LeonHeidelbach** [comment, marked as answer]
@luiz00martins Please read the documentation. That’s what it’s there for 😉. You will find every available command in the [user commands](https://github.com/LeonHeidelbach/trailblazer.nvim#-user-commands) section. The commands you are looking for are `:TrailBlazerPeekMovePreviousUp` and `:TrailBlazerPeekMoveNextDown`. If you have a look at the [configuration](https://github.com/LeonHeidelbach/trailblazer.nvim#%EF%B8%8F-configuration) section, you will find under `mappings` all default key maps for motions and actions which you can easily customize to your needs. I hope this helps.

**Reply by luiz00martins** [reply]
I think either `:TrailBlazerPeekMovePreviousUp` and `:TrailBlazerPeekMoveNextDown` do not do what I'm asking for, or they are bugged in my setup.

Below you can see and example where `:TrailBlazerPeekMovePreviousUp` and `:TrailBlazerPeekMoveNextDown` do not move to the mark, but `:TrailBlazerTrackBack` does:

https://user-images.githubusercontent.com/43142209/216326193-0529da63-974c-4bc5-a8c5-8d9c304fe953.mp4

**Reply by LeonHeidelbach** [reply]
@luiz00martins This might be caused by your setup (maybe the floating commandline), I cannot reproduce this issue. Have you tried using the key bindings to move through the stack? If not try using those. You could also try to disable the floating commandline plugin and see if using the commands works now.

https://user-images.githubusercontent.com/55194243/216328554-6a4d3a6d-c201-434c-b58d-53c767905e60.mp4

**Reply by LeonHeidelbach** [reply]
@luiz00martins I think I know what is causing this. Apparently using the commands does not work if you switch to a different window. This will be connected to the default parameter being the current window if you do not provide any arguments. I will look into this.

**Reply by LeonHeidelbach** [reply]
@luiz00martins I just pushed a fix for this. It should work now. Can you confirm? The issue actually was the default command value being an empty string. Good catch :+1: .

**Reply by luiz00martins** [reply]
Yep, it's working now! Thank you for the response!
